poe-api/src/poe.py

```diff
@@ -508,6 +508,59 @@ def get_message_history(self, chatbot, count=25, cursor=None):
     messages = query_messages + messages
     return messages
 
+  def get_latest_messages(self, chatbot, count=25, cursor=None):
+    logger.info(f"Downloading {count} messages from {chatbot}")
+
+    messages = []
+    if cursor == None:
+      chat_data = self.get_bot(self.bot_names[chatbot])
+      if not chat_data["messagesConnection"]["edges"]:
+        return []
+      messages = chat_data["messagesConnection"]["edges"][-count:][::-1]
+      cursor = chat_data["messagesConnection"]["pageInfo"]["startCursor"]
+      count -= len(messages)
+
+    cursor = str(cursor)
+    if count > 50:
+      messages = self.get_message_history(chatbot, count=50, cursor=cursor)[::-1] + messages
+      while count > 0:
+        count -= 50
+        new_cursor = messages[0]["cursor"]
+        new_messages = self.get_message_history(chatbot, min(50, count), cursor=new_cursor)[::-1]
+        messages = new_messages + messages
+      return messages
+    elif count <= 0:
+      return messages
+
+    result = self.send_query("ChatListPaginationQuery", {
+      "count": count,
+      "cursor": cursor,
+      "id": self.bots[chatbot]["id"]
+    })
+    query_messages = result["data"]["node"]["messagesConnection"]["edges"][::-1]
+    messages = query_messages + messages
+    return messages
+
+  def get_all_messages(self, chatbot, cursor=None):
```

Comment:
Wouldn't this not work at all since you're only calling `self.get_message_history` once?

Also, there's no need for a cursor argument here, and the `get_message_history` function already has code to figure out the correct cursor.